plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseHiveConnectorTest.java

```diff
@@ -5588,6 +5588,60 @@ private void testSchemaMismatchesWithDereferenceProjections(Session session, Hiv
         }
     }
 
+    @Test
+    public void testReadWithPartitionSchemaMismatch()
+    {
+        testWithAllStorageFormats(this::testReadWithPartitionSchemaMismatch);
+    }
+
+    private void testReadWithPartitionSchemaMismatch(Session session, HiveStorageFormat format)
+    {
+        String tableName = "read_with_partition_schema_mismatch_" + randomNameSuffix();
+        try {
+            if (isMappingByName(format)) {
```

Comment:
Can at least table setup be extracted a scommon (you have a,b,_part vs a,b,c,_part now)
would make it clearer what's the difference between by-name vs by-index mappings

i guess the test output doesn't change with the commit, so ;et's have it as  a prep commit

Reply:
restructured.